### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by kayintveen** [beyondcode/herd-community#382]
### Operating system version

macOS Sonoma 14.2.1 (23C71)

### System architecture

ARM64 (M1, M2, etc)

### Herd Version

1.4.0

### PHP Version

PHP 8.1.26 (cli) (built: Dec 21 2023 08:51:01) (NTS)

### Bug description

When I activated Pro, my CLI stopped working.

Warning: stream_resolve_include_path(): Unable to find the wrapper "phar" - did you forget to enable it when you configured PHP? in phar:///Applications/Herd.app/Contents/Resources/valet/dump.phar/vendor/composer/ClassLoader.php on line 557

Tried restarting, rebooting, herd restart and switching between PHP 8.1 (which I need for this project) and 8.2

### Steps to reproduce

actived pro on a clean install

**Comment by mpociot** [comment]
Are you sure that `php` on the CLI points to the correct binary?
Can you share your `which php` output?

**Comment by kayintveen** [comment]
Yes

❯ which php
/Users/kayintveen/Library/Application Support/Herd/bin//php

**Comment by mpociot** [comment]
What do you get when you run 

```shell
php -i | grep "Registered PHP Streams"
```

It should give you: "Registered PHP Streams => https, ftps, compress.zlib, compress.bzip2, php, file, glob, data, http, ftp, **phar**, zip"

**Comment by sschlein** [comment]
You should be back to a running system if you open the Dumps window and stop dump interception.

We're investigating this at the moment but it could be Magento related.

**Comment by kayintveen** [comment]
> What do you get when you run
> 
> ```shell
> php -i | grep "Registered PHP Streams"
> ```
> 
> It should give you: "Registered PHP Streams => https, ftps, compress.zlib, compress.bzip2, php, file, glob, data, http, ftp, **phar**, zip"

Phar is in there.
Registered PHP Streams => https, ftps, compress.zlib, compress.bzip2, php, file, glob, data, http, ftp, phar, zip





> You should be back to a running system if you open the Dumps window and stop dump interception.
> 
> We're investigating this at the moment but it could be Magento related.

i was not sure how to do this. I expected this to be in the dumps pane to enable / disable it. but its in the dumps screen itself I can stop listening. and this fixed it for now.

**Comment by mpociot** [comment]
This seems to be related to Magento unregistering the phar stream...

See: https://github.com/magento/magento2/issues/21973

I suppose you run into this when running Magento related commands via the CLI?

**Comment by mpociot** [comment]
Yeah, Herd Pro loads a phar file to inject the `dump` and `dd` methods, which is why you ran into it - and which is why disabling the dump interception fixes it.

**Comment by kayintveen** [comment]
For now i keep it disabled for magento hopefully a minor version upgrade will fix it later. Thanks!

**Comment by barryvdh** [comment]
Same issue with the latest version. How can I disable the dump interception?

**Comment by barryvdh** [comment]
Nevermind, find the button. But still breaks Magento

**Comment by mpociot** [comment]
@barryvdh I wasn't able to verify the fix that I sent to @kayintveen 
Could you try the following:

Add

```
if (!in_array('phar', \stream_get_wrappers())) {
    \stream_wrapper_restore('phar');
}
```

To this file: `/Applications/Herd.app/Contents/Resources/valet/dump-loader.php`

Right before the include.
It seems like Magento is still unregistering the phar stream

**Comment by barryvdh** [comment]
Not sure how to best test that, my Mac doesn't seem to like me changing that file.

**Comment by kayintveen** [comment]
Some how I fixed it. But I don't remember how. would love to have this working out of the box. i mainly had the issue with Magento as well. 
Let me check with my team as more people use Magento with Herd now.

**Comment by barryvdh** [comment]
If you disable the interception, there is no issue:
![image](https://github.com/beyondcode/herd-community/assets/973269/b397f2c7-7a4c-4842-9076-eb8118ebb55f)

Does intercepting work for you now?

I think the fix should work, it's also mentioned here; https://github.com/magento/magento2/issues/21973#issuecomment-783437166

**Comment by mpociot** [comment]
Yeah, when the interception is disabled we're not loading a PHAR file.

I'll just apply the fix in the next update, as I also doesn't see any reasons why this should break anything

**Comment by kayintveen** [comment]
Hi @mpociot I just re-subscribed to pro as it got expired. and again have a similar issue. thought I just mentioned it here as I was googling and stumbled upon my own issue ;-) 

```
Exception #0 (Exception): Warning: stream_resolve_include_path(): Unable to find the wrapper &quot;phar&quot; - did you forget to enable it when you configured PHP? in phar:///Applications/Herd.app/Contents/Resources/valet/dump.phar/vendor/composer/ClassLoader.php on line 551
```

**Comment by sebfie** [comment]
I still have this issue
